x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryBuilderResolver.java

```diff
@@ -69,12 +69,29 @@ private static Set<String> indexNames(LogicalPlan plan) {
         return indexNames;
     }
 
+    private static boolean hasFullTextFunctions(LogicalPlan plan) {
+        return plan.anyMatch(p -> {
+            Holder<Boolean> hasFullTextFunction = new Holder<>(false);
+            p.forEachExpression(FullTextFunction.class, unused -> hasFullTextFunction.set(true));
+
+            if (p instanceof Fork fork) {
```

Comment:
Should `Fork` implement `forEachExpression` instead? That would make things easier for the clients that will not need to take it into account, and be in line with the `Composite` pattern...